shared/router-v2/router.shared.js

```diff
@@ -67,49 +70,45 @@ export const oldActionToNewActions = (action: any, navigation: any) => {
     case RouteTreeGen.switchRouteDef: {
       // used to tell if its the login one or app one. this will all change when we deprecate the old routing
       const routeName = action.payload.routeDef.defaultSelected === 'tabs.loginTab' ? 'loggedOut' : 'loggedIn'
-      const switchStack = NavigationActions.navigate({params: undefined, routeName})
 
       // You're logged out
       if (routeName === 'loggedOut') {
-        return [switchStack]
+        return [NavigationActions.navigate({params: undefined, routeName: 'loggedOut'})]
       }
 
       // When we restore state we want the following stacks
-      // [People, TheLastTabYouWereOn, MaybeAConversationIfTheLastTabYouWereOnIsChat]
-      let sa = [StackActions.push({params: undefined, routeName: 'tabs.peopleTab'})]
+      let sa = [NavigationActions.navigate({params: undefined, routeName: 'loggedIn'})]
 
       if (action.payload.path) {
         const p = action.payload.path.last
           ? action.payload.path.last()
           : action.payload.path[action.payload.path.length - 1]
 
-        // a chat, we want people/inbox/chat
+        // a chat, we want inbox/chat
         if (p === 'chatConversation') {
-          sa.push(StackActions.push({params: undefined, routeName: 'tabs.chatTab'}))
+          sa.push(NavigationActions.navigate({params: undefined, routeName: 'tabs.chatTab'}))
```

Comment:
this will pop up to the tabs.chatTab, right?

Reply:
this switches the tabs